### ISSUE TITLE
False UndefinedVariable $query in #[Scope] method body when it overrides an abstract/interface scope

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1037]
## Summary

A `#[Scope]`-attributed scope method that **overrides an abstract (or interface) scope declaration** triggers a false `UndefinedVariable` (`Cannot find referenced variable $query`) **inside the scope method's own body**, where `$query` is the method's first parameter.

The plugin's scope-parameter stripping (declared params minus the leading `$query`, intended for *forwarded* call sites like `Model::scope()` / `$builder->scope()`) leaks into the override method's own body analysis, so `$query` never enters `vars_in_scope`.

## Minimal reproduction

```php
use Illuminate\Database\Eloquent\Attributes\Scope;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;

trait HasWorkflow
{
    abstract protected function wherePublished(Builder $query): void;
}

class Article extends Model
{
    use HasWorkflow;

    /** @param Builder<self> $query */
    #[\Override]
    #[Scope]
    protected function wherePublished(Builder $query): void
    {
        // false positive: UndefinedVariable - Cannot find referenced variable $query
        $query->where('status', 'published');
    }
}
```

The abstract declaration can live on a trait (as above) or an interface; both reproduce. An ordinary `#[Scope]` method that does **not** override anything is clean — the trigger is the override resolution path.

## Trigger signature (observed)

Perfect correlation across all affected sites:

* method has `#[Scope]`, **and**
* method has `#[\Override]` (i.e. it overrides an abstract/interface declaration of the same scope).

Non-overriding `#[Scope]` methods (e.g. a plain `withHigherRank(Builder $query, ...)`) do not reproduce.

## Likely cause

`BuilderScopeHandler::getScopeParams()` returns the scope's declared params **minus the leading `$query`** (`array_slice($codebase->methods->getMethodParams($scopeMethodId), 1)`), and `ModelMethodHandler::getMethodParams` (registered on the model class) serves those stripped params for `Model::scope()` / `$builder->scope()` call sites.

When the scope method **overrides** an abstract/interface method, Psalm resolves the method through the declaring-method path that consults the params provider, and the stripped (zero-`$query`) signature is applied to the override method's **own body analysis** — dropping the `$query` parameter from scope. `isDirectScopeCall()` guards the sibling direct-call case (`$this->otherScope($query, ...)`), but not the declaration body of the scope itself.

## Suggested fix

Never apply scope-param stripping to the scope method's **own** `FunctionLikeStorage->params` (body analysis); restrict stripping to the synthetic Builder/Model *call* signatures. Gate the params provider so it does not answer for the method when the analysis target is the declaring class of the method being analyzed (its own body), only when the call receiver is a `Builder`/`Model`.

## Environment

* psalm/plugin-laravel: `dev-master` (`f566c421`)
* vimeo/psalm: `7.0.0-beta19`
* Laravel 12 (attribute-based local scopes, `#[Scope]`)

**Comment by alies-dev** [comment]
Fixed on `master` by #1041.

The root cause was the scope params provider answering during the method's *own declaration* analysis: Psalm's `FunctionLikeAnalyzer` requests params for any overriding method (which is why `#[Override]` correlated perfectly — abstract trait/interface declarations populate `overridden_method_ids`), and the provider served the stripped (query-less) signature, so `$query` never entered `vars_in_scope`. #1041 replaced the argument-shape heuristic with dispatch-based classification and the provider now declines for declaration-analysis lookups.

Verified against `016b0d3c` with this issue's exact reproduction (abstract trait declaration + `#[Override]` + `#[Scope]`): no `UndefinedVariable`, and the interface-declaration variant resolves through the same path. The fix ships in v4.13.0.

Note: at `errorLevel="1"` the reproduction still reports `MoreSpecificImplementedParamType` for the `@param Builder<self>` docblock narrowing the abstract `Builder` signature — that one is accurate (the override declares a more specific param than the contract), unrelated to this bug, and fixable by annotating the abstract declaration too.
